packages/adapters/ethers/src/client.ts

```diff
@@ -631,9 +631,12 @@ export class EthersAdapter {
       if (provider) {
         const { addresses, chainId } = await EthersHelpersUtil.getUserInfo(provider)
         const firstAddress = addresses?.[0]
-        const caipAddress = `${this.chainNamespace}:${chainId}:${firstAddress}` as CaipAddress
+        const caipNetwork = this.caipNetworks.find(c => c.id === chainId) ?? this.caipNetworks[0]
+        const caipAddress =
+          `${this.chainNamespace}:${caipNetwork?.id}:${firstAddress}` as CaipAddress
```

Comment:
Have you tried switching from solana to evm? doesn't that break this cause of the "hardcoded" `this.chainNamespace` ?